src/app/views/canvas/chart_editor.tsx

```diff
@@ -596,6 +609,27 @@ export class ChartEditorView
     }
   }
 
+  public doCustomZoom(cx: number, cy: number, width: number, height: number) {
+    const width_main = this.state.viewWidth;
+    const height_main = this.state.viewHeight;
+
+    const newCX = width_main / 2 - cx;
+    const newCY = height_main / 2 + cy;
+
+    const newScale =
+      width_main > height_main ? height_main / height : width_main / width;
+
+    this.setState({
```

Comment:
`setState` is called here and then within `doZoom` - consider refactoring `doZoom` to include optional `cx` and `cy` parameters so that the translation of the view's center position and the zoom are all set by one `setState` call inside of `doZoom`.

Reply:
Moved to `doZoom`